cpp/journal_publishing_processor.cc

```diff
@@ -124,7 +129,26 @@ bool ExtractText(XMLParser * const xml_parser, const std::string &text_opening_t
 }
 
 
-void ProcessDocument(const bool normalise_only, const std::string &input_file_path, XMLParser * const xml_parser,
+void ExtractPDFFulltext(const bool force_ocr, const std::string &fulltext_location, std::string * const full_text) {
+    if (not StringUtil::EndsWith(fulltext_location, ".pdf", true))
```

Comment:
Please always add a comment in front of cryptic constants like "true" or "false".  In this case I'd write /* ignore_case = */.